contracts/SimpleOperations.sol

```diff
@@ -0,0 +1,161 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pragma solidity ^0.4.19;
+
+import "./Operations.sol";
+
+contract SimpleOperations is Operations {
+	struct Release {
+		uint32 forkBlock;
+		uint8 track;
+		uint24 semver;
+		bool critical;
+		mapping (bytes32 => bytes32) checksum; // platform -> checksum
+	}
+
+	struct Build {
+		bytes32 release;
+		bytes32 platform;
+	}
+
+	struct Client {
+		address owner;
+		mapping (bytes32 => Release) release;
+		mapping (uint8 => bytes32) current;
+		mapping (bytes32 => Build) build; // checksum -> Build
+	}
+
+	mapping (bytes32 => Client) public client;
+	mapping (address => bytes32) public clientOwner;
+
+	address public grandOwner = msg.sender;
+
+	event Received(address indexed from, uint value, bytes data);
+	event ReleaseAdded(bytes32 indexed client, uint32 indexed forkBlock, bytes32 release, uint8 track, uint24 semver, bool indexed critical);
+	event ChecksumAdded(bytes32 indexed client, bytes32 indexed release, bytes32 indexed platform, bytes32 checksum);
+	event ClientAdded(bytes32 indexed client, address owner);
+	event ClientRemoved(bytes32 indexed client);
+	event ClientOwnerChanged(bytes32 indexed client, address indexed old, address indexed now);
+	event OwnerChanged(address old, address now);
+
+	function SimpleOperations() public {
+		client["parity"] = Client(msg.sender);
+		clientOwner[msg.sender] = "parity";
+	}
+
+	function() public payable { Received(msg.sender, msg.value, msg.data); }
+
+	// Functions for client owners
+
+	function setClientOwner(address _newOwner) public only_client_owner {
+		var newClient = clientOwner[msg.sender];
+		clientOwner[msg.sender] = bytes32(0);
+		clientOwner[_newOwner] = newClient;
+		client[newClient].owner = _newOwner;
+		ClientOwnerChanged(newClient, msg.sender, _newOwner);
+	}
+
+	function addRelease(bytes32 _release, uint32 _forkBlock, uint8 _track, uint24 _semver, bool _critical) public only_client_owner {
+		var newClient = clientOwner[msg.sender];
+		client[newClient].release[_release] = Release(_forkBlock, _track, _semver, _critical);
+		client[newClient].current[_track] = _release;
+		ReleaseAdded(newClient, _forkBlock, _release, _track, _semver, _critical);
+	}
+
+	function addChecksum(bytes32 _release, bytes32 _platform, bytes32 _checksum) public only_client_owner {
+		var newClient = clientOwner[msg.sender];
+		client[newClient].build[_checksum] = Build(_release, _platform);
+		client[newClient].release[_release].checksum[_platform] = _checksum;
+		ChecksumAdded(newClient, _release, _platform, _checksum);
+	}
+
+	// Admin functions
+
+	function addClient(bytes32 _client, address _owner) public only_owner {
+		client[_client].owner = _owner;
```

Comment:
Should have a check that prevents calling if `_client` already has owner or `_owner` is already an owner of some other client.

Alternatively should be merged with `resetClientOwner` function that seems to be doing similar stuff.